src/imports.rs

```diff
@@ -55,6 +55,8 @@ fn rewrite_single_use_list(path_str: String, vpi: &ast::PathListItem) -> String
         // A name.
         if path_str.is_empty() {
```

Comment:
I guess if you are returning an empty path string, then you are hitting this path. So, we are not distinguishing between `::foo` and `foo`. I think the solution is not to branch on whether the path_str is empty (which could be due to either case) but change path_str to Option<String> where it is None for `foo` and Some("") for `::foo` (and it will need a comment explaining that).